test/functional/unified-spec-runner/runner.ts

```diff
@@ -41,7 +46,27 @@ export async function runUnifiedTest(
     ctx.skip();
   }
 
-  const utilClient = ctx.configuration.newClient();
+  let utilClient;
+  if (ctx.topologyType === uni.TopologyType.loadBalanced) {
+    // In load balanced mode we require these 2 environment
+    // variables to be set pointing at the load balancer
+    // frontends.
+    expect(SINGLE_MONGOS_LB_URI).to.exist;
+    expect(MULTI_MONGOS_LB_URI).to.exist;
+
+    // We determine which URI the load balancer to point at
+    // by looking at the MONGODB_URI - if it has 1 mongos
+    // then it's the single mongos frontend and if it has
+    // multiple mongoses then we use the multi mongos
+    // frontend.
+    if (isMultiHostUri(process.env.MONGODB_URI)) {
+      utilClient = ctx.configuration.newClient(MULTI_MONGOS_LB_URI);
+    } else {
+      utilClient = ctx.configuration.newClient(SINGLE_MONGOS_LB_URI);
+    }
+  } else {
+    utilClient = ctx.configuration.newClient();
+  }
 
   let entities;
   try {
```

Comment:
I attempted running the tests locally here's a summary of the issues I ran into:

- When I try to get the tests running I run into the serviceId missing error, we need a way to mock the serviceId on ismaster responses, I think other driver's have been using some global flag as a hack.
- I hardcoded the serviceId and the load-balanced topoplogy is missing from the `topologySatisfies` check in `test/functional/unified-spec-runner/unified-utils.ts`
- The `createFindCursor` operation is missing, We are going to need to support the changes made here: https://github.com/mongodb/specifications/commit/ed4e62b48939378b814cbd7d9d4939a2c1076a6b

Reply:
`createFindCursor` is now brought back as well as the `iterateUntilDocumentOrError` changes to handle both change streams and find cursors. `close` is also back as an operation as well as `assertNumberConnectionsCheckedOut`. `ignoreResultAndError` assertion is also brought back.